src/CoreWCF.Primitives/src/CoreWCF/Security/Tokens/SecurityContextCookieSerializer.cs

```diff
@@ -0,0 +1,307 @@
+            XmlHelper.WriteElementContentAsInt64(writer, dictionary.ExpiryTime, dictionary.EmptyString, tokenExpirationTime.ToUniversalTime().Ticks);
+            XmlHelper.WriteElementContentAsInt64(writer, dictionary.KeyEffectiveTime, dictionary.EmptyString, keyEffectiveTime.ToUniversalTime().Ticks);
+            XmlHelper.WriteElementContentAsInt64(writer, dictionary.KeyExpiryTime, dictionary.EmptyString, keyExpirationTime.ToUniversalTime().Ticks);
+
+            AuthorizationContext authContext = null;
+            if (authorizationPolicies != null)
+                authContext = AuthorizationContext.CreateDefaultAuthorizationContext(authorizationPolicies);
+
+            if (authContext != null && authContext.ClaimSets.Count != 0)
+            {
+                DataContractSerializer identitySerializer = DataContractSerializerDefaults.CreateSerializer(typeof(IIdentity), this.knownTypes, int.MaxValue);
+                DataContractSerializer claimSetSerializer = DataContractSerializerDefaults.CreateSerializer(typeof(ClaimSet), this.knownTypes, int.MaxValue);
+                DataContractSerializer claimSerializer = DataContractSerializerDefaults.CreateSerializer(typeof(Claim), this.knownTypes, int.MaxValue);
+                SctClaimSerializer.SerializeIdentities(authContext, dictionary, writer, identitySerializer);
+
+                writer.WriteStartElement(dictionary.ClaimSets, dictionary.EmptyString);
+                for (int i = 0; i < authContext.ClaimSets.Count; i++)
+                {
+                    SctClaimSerializer.SerializeClaimSet(authContext.ClaimSets[i], dictionary, writer, claimSetSerializer, claimSerializer);
+                }
+                writer.WriteEndElement();
+            }
+
+            writer.WriteEndElement();
+            writer.Flush();
+
+            byte[] serializedContext = stream.ToArray();
+            return this.securityStateEncoder.EncodeSecurityState(serializedContext);
+        }
+
+
+        public SecurityContextSecurityToken CreateSecurityContextFromCookie(byte[] encodedCookie, UniqueId contextId, UniqueId generation, string id, XmlDictionaryReaderQuotas quotas)
+        {
+            byte[] cookie = null;
+
+            try
+            {
+                cookie = this.securityStateEncoder.DecodeSecurityState(encodedCookie);
+            }
+            catch (Exception e)
+            {
+                if (Fx.IsFatal(e))
+                {
+                    throw;
+                }
+                OnInvalidCookieFailure(SR.Format(SR.SctCookieBlobDecodeFailure), e);
+            }
+            SecurityContextSecurityToken sct = DeserializeContext(cookie, encodedCookie, id, quotas);
+            if (sct.ContextId != contextId)
+            {
+                OnInvalidCookieFailure(SR.Format(SR.SctCookieValueMissingOrIncorrect, "ContextId"));
```

Comment:
nameof(SecurityContextSecurityToken.ContextId), similarly for KeyGeneration below